src/modm/platform/timer/stm32/module.lb

```diff
@@ -110,8 +110,8 @@ def prepare(module, options):
     props["target"] = target = device.identifier
     props["types"] = set()
     # extended advanced control timer with 6 channels and trigger out 2
-    props["advanced_extended"] = target.family not in ["f0", "f1", "f2", "f4", "l0", "l1"] or \
-            target.family == "f3" and target.name not in ["73", "78"]
+    props["advanced_extended"] = \
+            (target.family == "f3" and target.name not in ["03", "73", "78"])
```

Comment:
Nope. This breaks my production code :wink:. The advanced control timer implementation with 6 channels is present in all F7, G0, G4, H7, L4, L5, U5, WB and WL devices as well as all F3 except 373 and 378.

If you remove the advanced extended property user code using timer channels 5 and 6 will still compile and access wrong places in the peripheral memory when trying to configure those channels. That would be very bad there, for instance:

```
	static inline void
	setCompareValue(uint32_t channel, Value value)
	{
%% if advanced_extended
		if(channel <= 4) {
			*(&TIM{{ id }}->CCR1 + (channel - 1)) = value;
		} else if(channel == 5) {
			TIM{{ id }}->CCR5 = value;
		} else if(channel == 6) {
			TIM{{ id }}->CCR6 = value;
		}
%% else
		*(&TIM{{ id }}->CCR1 + (channel - 1)) = value;
%% endif
	}
```

Reply:
From looking at the ST headers the correct condition should be:
```suggestion
    props["advanced_extended"] = target.family not in ["f0", "f1", "f2", "f3", "f4", "l0", "l1"] and \
            (target.family != "f3" or target.name not in ["73", "78"])
```

If there is a compile error for F303, then all what we call `advanced_extended` is actually identical.

Reply:
... but GPIO `base.hpp` is missing `Ch5` and `Ch6` in many cases:

```
/// @cond
	enum class
	Signal
	{
[...]
		Ch1,
		Ch1n,
		Ch2,
		Ch2n,
		Ch3,
		Ch3n,
		Ch4,
		Ch4n,
		Cha1,
		Cha2,
		Chb1,
[...]
```

e.g. for `examples/nucleo_g474re/servo_pwm`
```
Compiling C++·· release/modm/src/modm/platform/timer/timer_4.o
In file included from modm/src/modm/platform/timer/timer_4.hpp:23,
                 from modm/src/modm/platform/timer/timer_4.cpp:17:
modm/src/modm/platform/timer/general_purpose_base.hpp: In static member function 'static consteval int modm::platform::GeneralPurposeTimer::signalToChannel()':
modm/src/modm/platform/timer/general_purpose_base.hpp:272:56: error: 'Ch5' is not a member of 'modm::platform::Gpio::Signal'; did you mean 'Ch1'?
  272 |   } else if constexpr (Signal::Signal == Gpio::Signal::Ch5) {
```

Reply:
The CI compile error is caused by the absence of `Gpio::Signal::Ch5` and `Gpio::Signal::Ch6`. Channels 5 and 6 can't be connected to IO pins on any STM32, so there are no GPIO signals for it. That was my mistake. Just remove the following lines in `signalToChannel()` and everything should work:

```
%% if advanced_extended
		} else if constexpr (Signal::Signal == Gpio::Signal::Ch5) {
			return 5;
		} else if constexpr (Signal::Signal == Gpio::Signal::Ch6) {
			return 6;
%% endif
```

Reply:
> Channels 5 and 6 can't be connected to IO pins on any STM32, so there are no GPIO signals for it.

I was just studying the datasheets and came to the same conclusion. No worries, just fixed it.